tests/common/devices/duthosts.py

```diff
@@ -56,15 +57,90 @@ def __init__(self, ansible_adhoc, tbinfo, duts):
         """
         self.ansible_adhoc = ansible_adhoc
         self.tbinfo = tbinfo
+        self.request = request
         self.duts = duts
-        self.__initialize_nodes()
+        self.is_parallel_run = target_hostname is not None
+        # TODO: Initialize the nodes in parallel using multi-threads?
+        if self.is_parallel_run:
+            self.parallel_run_stage = "non_initial_checks"
+            self.target_hostname = target_hostname
+            self.is_parallel_leader = is_parallel_leader
+            self.__initialize_nodes_for_parallel()
+        else:
+            self.__initialize_nodes()
```

Comment:
When parallel run is enabled, we want only the leader host to run the initial checks (sanity_check and core_dump_and_config_check). We know that both sanity_check and core_dump_and_config_check will call the `duthosts` fixture, but since this is parallel run, `duthosts` should only return a list containing the target host.

Therefore, the purpose of the changes here in `DutHosts` class is to resolve the above issue by dynamically rendering the value of `duthosts` fixture when parallel run is enabled. Specifically, we want `self.nodes` to return all the DUTs at 3 stages for the leader host:

- at very beginning of the test so fixtures depending on `duthosts` can also be initialized with all DUTs information, for example, the `creds_all_duts` fixture
- during initial checks setup
- during intial checks teardown

Otherwise, we want `self.nodes` to only return the target host for both the leader and followers.

FYI, the logic has not been changed when parallel run is disabled.